paper-server/src/main/java/io/papermc/paper/registry/data/PaperDamageTypeRegistryEntry.java

```diff
@@ -0,0 +1,121 @@
+package io.papermc.paper.registry.data;
+
+import io.papermc.paper.registry.PaperRegistryBuilder;
+import io.papermc.paper.registry.data.util.Conversions;
+import net.minecraft.world.damagesource.*;
+import org.bukkit.craftbukkit.damage.CraftDamageEffect;
+import org.bukkit.craftbukkit.damage.CraftDamageType;
+import org.bukkit.damage.DamageEffect;
+import org.jspecify.annotations.Nullable;
+
+import static io.papermc.paper.registry.data.util.Checks.asConfigured;
+
+public class PaperDamageTypeRegistryEntry implements DamageTypeRegistryEntry {
+    protected @Nullable String messageId;
+    protected float exhaustion;
```

Comment:
This should be nullable, and then the getter and `build` method should check that it's been set. It's not an optional field, so we want people to explicitly set it.